src/dex/aave-v3-stata/aave-v3-stata.ts

```diff
@@ -0,0 +1,507 @@
+  //   destToken: string,
+  //   srcAmount: string,
+  //   destAmount: string,
+  //   data: AaveV3StataData,
+  //   side: SwapSide,
+  // ): Promise<SimpleExchangeParam> {
+  //   const { exchange, srcType, destType } = data;
+  //   let swapData;
+  //
+  //   if (side === SwapSide.SELL) {
+  //     if (srcType === TokenType.STATA_TOKEN) {
+  //       // e.g. sell srcAmount 100 srcToken stataUSDC for destToken USDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.redeem, [
+  //         srcAmount,
+  //         this.augustusAddress, // receiver
+  //         this.augustusAddress, // owner
+  //         destType === TokenType.UNDERLYING, // withdraw from aToken
+  //       ]);
+  //     } else {
+  //       // sell srcAmount 100 srcToken USDC for destToken stataUSDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(
+  //         StataFunctions.deposit,
+  //         [
+  //           srcAmount,
+  //           this.augustusAddress, // receiver
+  //           0, // referrer (noop)
+  //           srcType === TokenType.UNDERLYING, // deposit to aave
+  //         ],
+  //       );
+  //     }
+  //   } else {
+  //     if (srcType === TokenType.STATA_TOKEN) {
+  //       // e.g. buy destAmount 100 destToken USDC for srcToken stataUSDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(
+  //         StataFunctions.withdraw,
+  //         [
+  //           destAmount,
+  //           this.augustusAddress, // receiver
+  //           this.augustusAddress, // owner
+  //         ],
+  //       );
+  //     } else {
+  //       // e.g. buy destAmount 100 destToken stataUSDC for srcToken USDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.mint, [
+  //         destAmount,
+  //         this.augustusAddress,
+  //       ]);
+  //     }
+  //   }
+  //
+  //   return this.buildSimpleParamWithoutWETHConversion(
+  //     srcToken,
+  //     srcAmount,
+  //     destToken,
+  //     destAmount,
+  //     swapData,
+  //     exchange,
+  //   );
+  // }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: AaveV3StataData,
+    side: SwapSide,
+    _: Context,
+    executorAddress: Address,
+  ): DexExchangeParam {
+    const { exchange, srcType, destType } = data;
+    let swapData;
+    let returnAmountPos = undefined;
+
+    if (side === SwapSide.SELL) {
+      if (srcType === TokenType.STATA_TOKEN) {
+        // e.g. sell srcAmount 100 srcToken stataUSDC for destToken USDC
+        swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.redeem, [
+          srcAmount,
+          recipient, // receiver
+          executorAddress, // owner
+          destType === TokenType.UNDERLYING, // withdraw from aToken
+        ]);
+
+        returnAmountPos = extractReturnAmountPosition(
+          AaveV3Stata.stata,
+          StataFunctions.redeem,
```

Comment:
You should pass function name, not function interface (`redeem(uint256,address,address,bool)`)

Reply:
Though it might work
<img width="987" alt="image" src="https://github.com/paraswap/paraswap-dex-lib/assets/82939501/b8d44eb2-521b-40a9-8ca6-1d3734c9cfad">

Reply:
There are 2 functions with the same name for both deposit & redeem, we're using the ones with 4 input params